docs/source/notes/autograd.rst

```diff
@@ -86,6 +36,150 @@ flow statements, that can change the overall shape and size of the graph at
 every iteration. You don't have to encode all possible paths before you
 launch the training - what you run is what you differentiate.
 
+.. _locally-disable-grad-doc:
+
+Locally disabling gradient computation
+--------------------------------------
+
+There are several mechanisms available from Python to locally disable gradient
+computation:
+
+To disable gradients across entire blocks of code, there are grad modes
+like no-grad mode and inference mode.
+For more fine-grained exclusion of subgraphs from gradient computation,
+there is setting the ``requires_grad`` field of a tensor.
+
+Below, in addition to discussing the mechanisms above, we also describe
+evaluation mode (:meth:``nn.Module.eval()``), a method that is not actually used
+to disable gradient computation but, because of its name, is often mixed up with the three.
+
+Setting ``requires_grad``
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:attr:`requires_grad` is a flag that allows for fine-grained exclusion of
+subgraphs from gradient computation. It takes effect in both the forward
+and backward passes:
+
+During the forward pass, an operation is only recorded in the backward graph if
+at least one of its input tensors require grad.
+During the backward pass (``.backward()``), only leaf tensors with
+``requires_grad=True`` will have gradients accumulated into their ``.grad``
+fields.
+
+It is important to note that even though every tensor has this flag,
+*setting* it only makes sense for leaf tensors (tensors that do not have a
+``grad_fn``, e.g., a module's parameters).
+Non-leaf tensors (tensors that do have ``grad_fn``) are actually defined to always
```

Comment:
I feel like these two sentences could be better. Maybe something like:

Non-leaf tensors (tensors that do have ``grad_fn``) are Tensors that have a backward
graph associated with them and so it is a Tensors whose gradient will be needed as an intermediary
result to compute the gradient for a leaf Tensor that requires grad.
From this definition, it is clear that all non-leaf Tensor will automatically have ``require_grad=True``.

Reply:
Nice, that is a lot clearer!